### DISCUSSION TITLE
Does RFT mode preserve names in the tracebacks?

### DISCUSSION THREAD

**Discussion by Thoufak** [dashingsoft/pyarmor#1698] [General]
If an exception is raised, will I see the changed names or the original names in the traceback when using RFT mode?

**Comment by jondy** [comment]
It's the changed names in the trace back. RFT mode will rewite the script in source level.

**Reply by Thoufak** [reply]
I see. This makes debugging hard. Do I understand correctly that all RFT does is renaming? And if I want to avoid renaming, using RFT doesn't make sense for me?

**Reply by Thoufak** [reply]
Also, a similar question about BCC. As it converts some functions to the C code, is the traceback for these functions messed up?

**Comment by jondy** [comment]
Please check topics part in the documentation: `insight into RFT mode` and `insight into BCC mode`
